src/transformers/trainer.py

```diff
@@ -553,7 +565,7 @@ def _get_train_sampler(self) -> Optional[torch.utils.data.sampler.Sampler]:
 
         else:
             if self.args.world_size <= 1:
-                return RandomSampler(self.train_dataset)
+                return RandomSampler(self.train_dataset, generator=generator)
```

Comment:
Won't this fail in older pytorch? as it has no `generator` in its sig?

https://pytorch.org/docs/1.4.0/_modules/torch/utils/data/sampler.html#RandomSampler

That's why I proposed to use kwargs solution in an earlier comment, which can be empty.

Reply:
Good catch!